security/integration/webserver/src/test/java/io/helidon/security/integration/webserver/WebSecurityTests.java

```diff
@@ -50,28 +51,31 @@
     static final String AUDIT_MESSAGE_FORMAT = "Unit test message format";
     static UnitTestAuditProvider myAuditProvider;
     static WebServer server;
-    private static Client authFeatureClient;
-    private static Client client;
+    private static WebClient securitySetup;
+    private static WebClient webClient;
 
     @BeforeAll
     static void buildClients() {
-        authFeatureClient = ClientBuilder.newClient()
-                .register(HttpAuthenticationFeature.universalBuilder().build());
-        client = ClientBuilder.newClient();
+        Security security = Security.builder()
+                .addProvider(HttpBasicAuthProvider.builder().build())
+                .build();
+
+        securitySetup = WebClient.builder()
+                .register(WebClientSecurity.create(security))
+                .build();
+
+        webClient = WebClient.create();
     }
 
     @AfterAll
     static void stopIt() throws InterruptedException {
-        authFeatureClient.close();
-        client.close();
-
         WebSecurityTestUtil.stopServer(server);
     }
 
     abstract String serverBaseUri();
 
     @Test
-    void basicTestJohn() {
+    void basicTestJohn() throws ExecutionException, InterruptedException {
```

Comment:
Maybe these could be changed to just `throws Exception` ? Or maybe like Tomas said we need to come-up with a small utility and wrap the exceptions.